@types/phaser3-rex-plugins/templates/ui/ui-plugin.d.ts

```diff
@@ -0,0 +1,27 @@
+declare module 'phaser3-rex-plugins/templates/ui/ui-plugin' {
+    import * as Phaser from 'phaser';
+    import ObjectFactory from 'phaser3-rex-plugins/templates/ui/ObjectFactory';
+    import GetViewport from 'phaser3-rex-plugins/plugins/utils/system/GetViewport';
+    import {GetParentSizer, GetTopmostSizer} from 'phaser3-rex-plugins/templates/ui/utils/GetParentSizer';
+    import {Hide, Show, IsShown} from 'phaser3-rex-plugins/templates/ui/utils/Hide';
+    import {WaitEvent, WaitComplete} from 'phaser3-rex-plugins/plugins/utils/promise/WaitEvent';
```

Comment:
Please, don't import them directly. Let's create a file `templates/ui/utils/WaitEvent.d.ts` and import them form this file.